Adding manual for Sonoff T1 UK 3 Gang V1.1 (#182)
OttoWinter/esphomedocs: devices/sonoff_t1_uk_3gang_v1.1.rst

Using With Sonoff T1 UK 3 Gang
==============================

.. seo::
    :description: Instructions for putting Sonoff T1 UK 3 Gang devices into flash mode and installing ESPHome on them.
    :image: sonoff_t1_uk_3g_v1.1.jpg

ESPHome can also be used with Sonoff T1 UK 3 Gang wireless switches. These devices are
basically just an ESP8266 chip with 3 relays to control power output ant three backlit capacitive touch buttons to control the relays.

.. note::

>     This instruction is made for T1 UK with the touch board v1.1 which is tricker to get into flash mode

Reviewer: ```suggestion
    This instruction is made for T1 UK with the touch board v1.1 which is trickier to get into flash mode
```

This PCB contains the ESP chip and what's left inside the switch body are relays.

.. figure:: images/sonoff_t1_uk_3g_plate_off_v1.1.jpeg
    :align: center
    :width: 75.0%

    "TOUCH BOARD" with touch pads holds the ESP chip.

Step 2: Connecting UART
-----------------------

Now we need our computer to somehow establish a data connection to the board. For this we will
have to connect the four wires on the UART to USB bridge to the UART pins of the Sonoff T1 v1.1.

Fortunately for us, these pins on the ESP controller have dedicated solder pads on the PCB. You can identify
these by the ``VCC33``, ``RX``, ``TX`` and ``GND`` markings on the silk-screen.

Now go ahead and connect these pins to your UART to USB bridge as seen in below image. Make sure
that you connect these correctly, especially the ``VCC33`` and ``GND`` parts as you can otherwise
destroy the chip.

``VCC33`` should be connected to the ``3V3`` (**not** 5V) pin of the UART bridge, ``GND`` to ``GND``
and the same with ``RX``/``TX``.

There's no need for soldering - for quick job like one time firmware flashing you can just hold pins in respective holes by hand

> providing ``you are not touching any live contacts``, only the wires.

Reviewer: ```suggestion
provided **you are not touching any live contacts**, only the wires.
```

pre-compile the firmware.

.. note::

    After this step, you will be able to find the compiled binary under
    ``<NAME_OF_NODE>/.pioenvs/<NAME_OF_NODE>/firmware.bin``. If you're having trouble with
    uploading, you can also try uploading this file directly with other tools.

Step 4: Uploading Firmware
--------------------------

In order to upload the firmware, you're first going to need to get the chip into a flash mode, otherwise
the device will start up without accepting any firmware flash attempts. 
To put ESP8266 into flash mode you need to connect ``GPIO0`` to ``GND`` when the device is powering up.

This is a tricky process with T1 and the best way to do it is to use a wire with pins on either side.
To do this, while the device is UART bridge is not connected to your USB port, flip the PCB over,
take a wire and connect the second Ground hole on the PCB (red) to the third from the right bottom leg on the chip as depicted below (yellow) -
that leg is connected to the GPIO0 on ESP and plug the UART to your USB port.

Keep holding  GND and GPIO0 connected for 2-4 seconds. The T1 UK 3 Gang should now be in a flash mode and should not blink with any LED.

> Touchpads may be lit up.

Reviewer: ```suggestion
The touchpads may light up.
```

.. figure:: images/sonoff_t1_uk_3g_back_plate_v1.1.jpeg
    :align: center

>     Connect red and yellow contacts while powering the boar up.

Reviewer: ```suggestion
    Connect red and yellow contacts while powering the board.
```
or 
```suggestion
    Connect red and yellow contacts while booting the board.
```

Now you can finally run the upload command:

.. code-block:: bash

    esphome sonoff_t1_uk_3g_v1.1.yaml run

If successful, you should see something like this:

.. figure:: images/sonoff_4ch_upload.png
    :align: center

Hooray 🎉! You've now successfully uploaded the first ESPHome firmware to your Sonoff T1 UK 3 Gang. And in a moment,
you will be able to use all of ESPHome's great features with your Sonoff T1 UK 3 Gang. Now you can put your T1 back together and fire up.

.. note::
    While now your T1 will start up and connect to your WiFi network if you power it up from UART it will not behave normally,
    it may flash random LEDs, turn on anf off touchpads' backlight and not react on touching touchpads. This will all be fixed once you re-assemble your T1

>     and powerit up from the mains power once safe to do so.

Reviewer: ```suggestion
    and power it up from the mains power once safe to do so.
```
